harness/determined/experimental/_native.py

```diff
@@ -110,14 +111,25 @@ def _make_test_workloads(config: det.ExperimentConfig) -> workload.Stream:
     logging.info("Finished saving a checkpoint.")
 
 
+def _parse_trial_class_from_entrypoint(entrypoint: str) -> Optional[str]:
+    entrypoint_cmd = entrypoint.split(" ")
+    trial_class_regex = re.compile("^[a-zA-Z0-9_.]+:[a-zA-Z0-9_]+$")
+    for arg in entrypoint_cmd:
+        if trial_class_regex.match(arg):
+            return arg
+    return None
+
+
 def _load_trial_for_checkpoint_export(
     context_dir: pathlib.Path,
     managed_training: bool,
     config: Dict[str, Any],
     hparams: Dict[str, Any],
 ) -> Tuple[Type[det.Trial], det.TrialContext]:
     with det._local_execution_manager(context_dir):
-        trial_class = load.trial_class_from_entrypoint(config["entrypoint"])
+        trial_class_entrypoint = _parse_trial_class_from_entrypoint(config["entrypoint"])
+        assert trial_class_entrypoint is not None
```

Comment:
If there is a chance that a users hits a "silent" assertion error (one without an explanation afterwards) during `Checkpoint.load()` then we need to either eliminate that chance or write an error message.

In this case, what will this do when you load a checkpoint that doesn't have the legacy entrypoint field?  I think writing an error message would be an insufficient solution; I think there are cases here that are just broken.

Reply:
as discussed, I'm going to save the trial class entrypoint specification in checkpoint metadata. once your checkpoint API changes are merged, I can make the changes. for now, launch layer works only with legacy model_def:Trial format.